src/test/cypress/integration/exam/ExamAssessment.spec.ts

```diff
@@ -0,0 +1,175 @@
+                courseManagementRequests.generateMissingIndividualExams(exam);
+                courseManagementRequests.prepareExerciseStartForExam(exam);
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                examStartEnd.startExam();
+                cy.contains(modelingResponse.body.title).should('be.visible').click();
+                modelingEditor.addComponentToModel(1);
+                modelingEditor.addComponentToModel(2);
+                modelingEditor.addComponentToModel(3);
+                cy.intercept(PUT, BASE_API + 'exercises/*/modeling-submissions').as('createModelingSubmission');
+                cy.contains('Save').click();
+                cy.wait('@createModelingSubmission');
+                cy.get('#exam-navigation-bar').find('.btn-danger').click();
+                examStartEnd.finishExam();
+                cy.login(tutor, '/course-management/' + course.id + '/exams');
+                cy.contains('Assessment Dashboard', { timeout: 60000 }).click();
+                assessmentDashboard.startAssessing();
+                modelingAssessment.addNewFeedback(2, 'Noice');
+                modelingAssessment.openAssessmentForComponent(1);
+                modelingAssessment.assessComponent(1, 'Good');
+                modelingAssessment.openAssessmentForComponent(2);
+                modelingAssessment.assessComponent(0, 'Neutral');
+                modelingAssessment.openAssessmentForComponent(3);
+                modelingAssessment.assessComponent(-1, 'Wrong');
+                assessmentDashboard.submitAssessment();
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                cy.get('.question-options').contains('2 of 10 points').should('be.visible');
+            });
+        });
+
+        it('Assess a text exercise submission', () => {
+            const exerciseTitle = 'Cypress Text Exercise';
+            courseManagementRequests.createAndAddTextExerciseToExam(exerciseGroup, exerciseTitle);
+            courseManagementRequests.generateMissingIndividualExams(exam);
+            courseManagementRequests.prepareExerciseStartForExam(exam);
+            cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+            examStartEnd.startExam();
+            cy.contains(exerciseTitle).click();
+            cy.get('#text-editor-tab').type(textSubmission.text);
+            cy.contains('Save').click();
+            cy.get('#exam-navigation-bar').find('.btn-danger').click();
+            examStartEnd.finishExam();
+            cy.login(tutor, '/course-management/' + course.id + '/exams');
+            cy.contains('Assessment Dashboard', { timeout: 60000 }).click();
+            assessmentDashboard.startAssessing();
+            assessmentDashboard.addNewFeedback(7, 'Good job');
+            assessmentDashboard.submitAssessment();
```

Comment:
When I execute the tests locally the POST request, which saves the assessment, is sometimes canceled because the test immediately logs out and logs in as a student. Waiting for the query to finish should help here.
```suggestion
            cy.intercept(POST, BASE_API + 'participations/*/results/*/submit-text-assessment').as('submitFeedback');
            assessmentDashboard.submitAssessment();
            cy.wait('@submitFeedback');
```